src/track/trackrecord.h

```diff
@@ -115,15 +115,20 @@ class TrackRecord final {
         /// The metadata has not been imported yet.
         Void,
 
-        /// The status could not be determined for whatever reason,
-        /// e.g. insufficient data, inaccessible file, ...
+        /// The metadata has been imported once, but until Mixxx 2.4 no
+        /// synchronization time stamps have been stored in the database
+        /// that allow to monitor the modification time of the file.
         Unknown,
 
         /// The metadata in Mixxx is up-to-date.
         Synchronized,
 
         /// The metadata in Mixxx is older than the metadata stored in file tags.
         Outdated,
+
+        /// The status could not be determined for whatever reason,
+        /// e.g. inaccessible file, ...
+        Undefined,
```

Comment:
The difference between Undefined/Unknown/Void is not obvious in the using code. 
How about renaming these value to 
NotImported/UnknownSyncronizedTime/UnknownFileTime

Reply:
The synchronization status type should not mention files directly. It is independent of how the content of sources is stored. Files are just one kind of storage type.

As already mentioned synchronization time stamps should not be confused with raw file time stamps and we should not mix that up on this higher level of abstraction.

The names of the variants are abstract and do not need to leak implementation details from other parts of the code. Descriptions of possible use cases have deliberately been added as possible examples in the comments.

I also dislike composed terms that start with the same stem. Different kind of "unknowns" are confusing, too. It is easier to recognize distinct words.

Reply:
Please provide a mapping to the existing values.

Reply:
Void = Unsynced
Syncronized = Syncronized
Outdated = ImportPending
Undefined = ExportPending

Reply:
Undefined = the source is inaccessible.

Reply:
Outdated is also somehow "unsynchronized". I don't think that this renaming improves anything.

Reply:
I think the names in the current state of this PR are okay, and in case something is unclear there are documentation comments in the header file.

Reply:
At least there is room for improvement for Undefined/Unknown/Void. In the best case the code should be self explaining without referring the header files for comment.  @uklotzde whats your proposal to fix this?

Reply:
My proposal is this PR. I did not receive any alternative proposals yet that I consider an improvement.

Reply:
For my feeling my recent proposal follows your demand in https://github.com/mixxxdj/mixxx/pull/4628#discussion_r785469202 
My demand is to have more descriptive names than the ambiguous Undefined/Unknown/Void.
What would be a  alternative for you?

Reply:
The variant names are recognizable and they were helpful while writing the different test cases.

They represent a state and are not directly related to any import/export activity as you suggested.

Reply:
If you really insist to keep these empty and interchangeable words Undefined/Unknown/Void and like to rely on out of sight comments, I'll give up. It is a pity that we have not found a compromise.

Reply:
Naming can be changed at any time. But I refuse to overload this PR with more unrelated changes.